src/core/DY.Core.Bytes.fst

```diff
@@ -1594,3 +1733,187 @@ let get_usage_dh_unknown_peer sk pk =
   normalize_term_spec get_usage;
   let DhKey sk_tag = get_usage sk in
   FStar.Classical.forall_intro (FStar.Classical.move_requires (dh_unknown_peer_usage_implies sk_tag))
+
+(*** KDF ***)
+
+/// Constructor.
+
+[@@"opaque_to_smt"]
+val kdf_extract: bytes -> bytes -> bytes
+let kdf_extract salt ikm =
+  KdfExtract salt ikm
+
+/// Constructor.
+
+[@@"opaque_to_smt"]
+val kdf_expand: bytes -> bytes -> len:nat{len <> 0} -> bytes
```

Comment:
I guess `len > 0` is implied by `len <> 0` and `len : nat`, but it still reads a bit strangely at first glance.

Reply:
Ah indeed. I guess we could use the type `pos` which is exactly that. Such `len <> 0` also happen in the length argument for random bytestrings.